### ISSUE TITLE
Cannot deliver emails to outlook.com address

### ISSUE THREAD

**Issue by figgyc** [foxcpp/maddy#301]
# Describe the bug
Just set up a maddy server on my domain today, was testing it by sending an email to `myemail@outlook.com` (obviously not my real email but you get the point) and it isn't working properly. Gmail works fine so it is probably a server specific problem. I can also recieve from Outlook just fine. The error says `TLS it not available or unauthenticated but required`, so for detail, I am using let's encrypt on my server to provide TLS.

# Steps to reproduce
Send a message to an outlook.com email, and check the logs

# Log files
https://hastebin.com/saqutukexi.apache (some details redacted)

# Configuration file
https://hastebin.com/dusubuvece.properties
(yes i use a catch-all on my email and I know the manual says not to do that but I did it on my old cloud email so if i reversed that now I would lose loads of email so yeah) Apart from that it's pretty much default but I figured to provide it anyway
# Environment information

* maddy version: 0.4.2+gbf982eb linux/amd64 go1.14.6 (self compiled)
* Debian 10

**Comment by foxcpp** [comment]
Not sure what's up with that but it looks like outlook.com MX does not support TLS (no STARTTLS advertised). The error you are getting is caused by this particular line in config:
```
mx_auth {
    dane
    mtasts {
        cache fs
        fs_dir mtasts_cache/
    }
    local_policy {
        min_tls_level encrypted #<---- this
        min_mx_level none
    }
}
```
It makes TLS mandatory for outbound connections. You can change the value to `none` to allow delivery without TLS use.

<details>
<summary>SMTP log showing no STARTTLS is advertised</summary>

```
[foxcpp@gamma ~]$ nc -C outlook-com.olc.protection.outlook.com. 25
220 DB8EUR06FT019.mail.protection.outlook.com Microsoft ESMTP MAIL Service ready at Fri, 27 Nov 2020 17:01:21 +0000
EHLO gamma.hexanet.dev
250-DB8EUR06FT019.mail.protection.outlook.com Hello [116.202.176.182]
250-SIZE 49283072
250-PIPELINING
250-DSN
250-ENHANCEDSTATUSCODES
250-8BITMIME
250-BINARYMIME
250-CHUNKING
250 SMTPUTF8
QUIT
```

</details>

**Comment by foxcpp** [comment]
Okay. Looks like only one (?) specific outlook.com MX server is lacking TLS support. I reported this via MS's deliverability support form.

**Comment by figgyc** [comment]
Thanks for the help. I reconfigured that settings but now I've noticed that my IP is blocked on their spam list :/ Maybe because it was failing to send so many times, or maybe just because I only started using this server today. Anyway the mail server is fine so I'm going to close the issue now.

**Comment by foxcpp** [comment]
Outlook.com MXes do not advertise STARTTLS to blacklisted IPs.

**Comment by dimm0** [comment]
I'm seeing the same error with the internet2.edu mail server. https://dnslytics.com/email-test test doesn't find any issues

Is there anything I can let them know to fix?

**Comment by reivilibre** [comment]
> Outlook.com MXes do not advertise STARTTLS to blacklisted IPs.

This also seems to be my experience with Hotmail. I had no STARTTLS support, and after removing the requirement for TLS, was told that my ISP is blacklisted.

However, e-mails have gone missing since the 10th of December and I only just found out the hard way last night (13th January), that this has been going on.

I haven't received any failure messages that informed me. Is it a better idea to open a new issue about this? I feel like it would have been good to have been alerted by this point.

**Comment by foxcpp** [comment]
>However, e-mails have gone missing since the 10th of December and I only just found out the hard way last night (13th January), that this has been going on.

I believe security policy errors currently defer delivery instead of failing it with a permanent error (and thus notifying you immediately). With default configuration it means messages get stuck in the queue for up to 5 days before actually giving up even if it is hopeless. The reason for this behavior is potential intermittent nature of various delivery failures. Sorry for your experience.

Do you think implementing a notification saying that the message is not delivered is a good solution to this?

**Comment by foxcpp** [comment]
Created a new issue for this problem.
